src/TimeModeling/Types/GeometryStructure.jl

```diff
@@ -172,10 +172,64 @@ end
 
 ################################################ Constructors from SEGY data  ####################################################
 
+# Utility function to prepare dtCell, ntCell, tCell from SEGY or based on user defined dt and t.
+# Useful when creating geometry for Forward Modeling with custom timings.
+function timings_from_segy(data::Union{SegyIO.SeisBlock,SegyIO.SeisCon,Array{SegyIO.SeisCon,1}}; dt=[], t=[])
+
+    # Get nsrc
+    if isa(data,SegyIO.SeisBlock)
+        src = get_header(data,"FieldRecord")
+        nsrc = length(unique(src))
+    else
+        nsrc = length(data)
+    end
+
+    # Prepare dtCell
+    if isempty(dt)
```

Comment:
This could probably be cleaned up with multiple dispatch but that's fine like that.